src/js/components/Dashboard.jsx

```diff
@@ -1,48 +1,45 @@
+import * as FrameworkTypes from 'safe-framework'
+import {ChartComponent} from './visualization/ChartComponent'
+import {ChartMenu} from './visualization/ChartMenu'
 import {connect} from 'react-redux'
 import {excludeEmptyFilters} from '../modules/utilities'
 import {fetchVisualizationResults} from '../modules/visualization-results'
 import FilterCriteria from './FilterCriteria'
 import {Hydrateable} from '../decorators/Hydrateable'
+import {MapComponent} from './visualization/MapComponent'
 import {setDefaultFilters} from '../modules/filters'
-import {verticalTop} from '../styles/common'
-import Visualization from './visualization/Visualization'
-import {GridList, GridTile} from 'material-ui/GridList'
+import {SummaryComponent} from './visualization/SummaryComponent'
+import {TableComponent} from './visualization/TableComponent'
+import {TableMenu} from './visualization/TableMenu'
 import React, {Component, PropTypes} from 'react'
 
-const style = {
-  gridList: {
-    width: '100%',
-    height: '60%',
-    overflowY: 'auto',
-    marginBottom: 24
-  },
-  gridTile: {
-    boxSizing: 'border-box',
-    height: '500px'
-  },
-  visualization: {
-    height: '100%'
-  }
-}
+const componentTypes = {ChartComponent, MapComponent, TableComponent, SummaryComponent}
+const menuTypes = {ChartMenu, TableMenu}
+
+const {Dashboard: FrameworkDashboard} = FrameworkTypes
 
-@Hydrateable('Dashboard', ['filters'], 'dashboard._id')
+@Hydrateable('Dashboard', ['filters'], 'dashboardId')
 class Dashboard extends Component {
   static propTypes = {
     dashboard: PropTypes.object.isRequired,
-    dashboards: PropTypes.object.isRequired,
+    dashboardId: PropTypes.string,
     dispatch: PropTypes.func.isRequired,
     filters: PropTypes.array.isRequired,
     visualizationResults: PropTypes.object.isRequired
   }
-
+  
   constructor (props) {
     super(props)
 
+    this.fetchVisualizationResults = ::this.fetchVisualizationResults
     this.onClickFilter = ::this.onClickFilter
+    this.geComponentType = ::this.geComponentType
```

Comment:
should be `this.getComponentType` instead of `this.geComponentType` (missing the t in get)